### PR TITLE
SES: tolerate RegExp() without new

### LINKED ISSUE #230: RegExp() (without new) fails under ses-0.7.3 unless noTameRegExp is enabled
I tried loading the `tap` unit-testing library in a SES-locked-down environment, and got an error I don't understand.

Clone [this gist repo](https://gist.github.com/warner/dfde9f05c3d80497ac9aa2c72493506a) and run `yarn` and `node -r esm main.js`. This imports one module (which imports `lockdown` and calls it), then imports a second module (which imports `tap`). The error I get (under node-13.7.0) is:

```
$ node -r esm main.js
did lockdown()

/tmp/dfde9f05c3d80497ac9aa2c72493506a/node_modules/ses/dist/ses.cjs:2779
    return Reflect.construct(unsafeRegExp, arguments, new.target);
                   ^
/tmp/dfde9f05c3d80497ac9aa2c72493506a/node_modules/ses/dist/ses.cjs:1
TypeError: undefined is not a constructor
    at RegExp (/tmp/dfde9f05c3d80497ac9aa2c72493506a/node_modules/ses/dist/ses.cjs:2779:20)
    at Object.<anonymous> (/tmp/dfde9f05c3d80497ac9aa2c72493506a/node_modules/tap/node_modules/yaml/dist/tags/yaml-1.1/timestamp.js:72:9)
    at Object.Module._compile (/tmp/dfde9f05c3d80497ac9aa2c72493506a/node_modules/source-map-support/source-map-support.js:541:25)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1171:10)
```

This may be a matter of `tap` not being compatible with SES, but 1) maybe we can figure out the problem and fix it, and 2) it's interfering with the #216 ses-adapter work. This pattern, where we lock everything down via an import at the beginning of the application, then import everything else, is our main plan for using `ses-adapter`.

I looked a little bit closer, and noticed that the `yaml` module in the stack trace is calling `RegExp` without a `new`.. not sure if that's significant or not:

```js
const timestamp = {
  identify: value => value instanceof Date,
  default: true,
  tag: 'tag:yaml.org,2002:timestamp',
  // If the time zone is omitted, the timestamp is assumed to be specified in UTC. The time part
  // may be omitted altogether, resulting in a date format. In such a case, the time part is
  // assumed to be 00:00:00Z (start of day, UTC).
  test: RegExp('^(?:' + '([0-9]{4})-([0-9]{1,2})-([0-9]{1,2})' + // YYYY-Mm-Dd
  '(?:(?:t|T|[ \\t]+)' + // t | T | whitespace
  '([0-9]{1,2}):([0-9]{1,2}):([0-9]{1,2}(\\.[0-9]+)?)' + // Hh:Mm:Ss(.ss)?
  '(?:[ \\t]*(Z|[-+][012]?[0-9](?::[0-9]{2})?))?' + // Z | +5 | -03:30
  ')?' + ')$'),
...
```

### PR DESCRIPTION
**Pull request by warner**
Somehow we were only testing `new RegExp()`, and not plain `RegExp()`. The latter was failing with an `undefined is not a constructor` error, which broke a lot of packages unless `noTameRegExp: true` was set.

Thanks to @erights for the recommended patch.

closes #230

### CODE REVIEW

**Comment by warner** [comment]
Before we consider this resolved, I'd like to know why this wasn't covered by our existing tests, particularly the test262 suite.

**Comment by warner** [comment]
Ah, @erights and I decided that test262's `built-ins/RegExp/call_with_regexp_not_same_constructor` ought to exercise this case, but that test (and about 50 others) are currently disabled in https://github.com/Agoric/SES-shim/blob/SES-v0.7.3/packages/ses/test262/tame-global-regexp-object.js#L13 . So this is just part of the incompleteness.

I'll see how many of those excluded tests now pass with this change.

**Comment by warner** [comment]
I was able to re-enable 21 test262 tests now that bare `RegExp()` works, as well as another 8 which worked even before that fix. We still have a lot of exclusions (587), but this is progress.

File: packages/ses/test/tame-global-reg-exp-object.test.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import tameGlobalRegExpObject from '../src/tame-global-reg-exp-object.js';
 const { test } = tap;
 
 test('tameGlobalRegExpObject - constructor', t => {
-  t.plan(6);
+  t.plan(8);
```
Comment (erights) [inline packages/ses/test/tame-global-reg-exp-object.test.js]: Why do we ***ever*** use `t.plan` rather than `t.end`?

Comment (warner) [reply packages/ses/test/tame-global-reg-exp-object.test.js]: I dunno.. it was like that when I found it :). I think JF had an argument for `t.plan` that made sense at the time, but I don't remember what it was.

**Pull request merged**